sql-parser/src/main/java/io/crate/sql/Identifiers.java

```diff
@@ -22,16 +22,30 @@
 
 package io.crate.sql;
 
+import io.crate.sql.parser.ParsingException;
 import io.crate.sql.parser.SqlParser;
+import io.crate.sql.parser.antlr.v4.SqlBaseLexer;
+import io.crate.sql.tree.QualifiedNameReference;
+import org.antlr.v4.runtime.Vocabulary;
 
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 public class Identifiers {
 
+    private static final Pattern IDENTIFIER = Pattern.compile("'([A-Z_]+)'");
     private static final Pattern ESCAPE_REPLACE_RE = Pattern.compile("\"", Pattern.LITERAL);
     private static final String ESCAPE_REPLACEMENT = Matcher.quoteReplacement("\"\"");
 
+    private static final Set<String> KEYWORDS = identifierCandidates().stream()
+        .filter(Identifiers::reserved)
+        .collect(Collectors.toSet());
```

Comment:
It is not really needed to use the stream API here. You can just pass the predicate as an argument to the `identifierCandidates` method to avoid allocating of an additional hashset and a pass over identifier candidates. it is a micro-optimization though, not sure whether it worth it.

Reply:
This is only run once in the life-cycle of a node since it's a static final field. So no it's not worth changing.